specs/honest-challenger-fdg.md

```diff
@@ -0,0 +1,145 @@
+instruction within their execution trace.
+2. **Do Nothing** if they agree with the root claim. If an honest challenger agrees with a
+root claim of a game, it means that they disagree with the output root it is trying to delete.
+They do nothing because if the root claim is left un-countered, the game will delete the output
+root they disagree with.
+
+### Counter Claims
+
+When a new claim is made in a dispute game with a [game tree](./fault-dispute-game.md#game-tree)
+depth in the range of `[1, MAX_DEPTH]`, the honest challenger processes it and performs
+a response. If multiple claims are observed since the last sync,
+the honest challenger responds to them in chronological order.
+
+To determine the appropriate response, the challenger first needs to know which
+[_team_](./fault-dispute-game.md#team-dynamics) it belongs to.
+This determines the set of claims it should respond to in the FDG.
+If the agent determines itself to be a Defender, aiming to delete an output root,
+ then it must dispute claims positioned at odd depths in the game tree.
+Otherwise, it disputes claims positioned at even depths in the game tree.
+This means an honest challenger will only respond to claims made by the opposing team.
+
+The next step is to determine whether the claim has a valid commitment (i.e. `ClaimHash`).
+If the `ClaimHash` matches ours at the same trace index, then we disagree with the claim's
+stance by moving to [defend](./fault-dispute-game.md#defend).
+Otherwise, the claim is [attacked](./fault-dispute-game.md#attack).
+
+The following pseudocode illustrates the response logic.
+
+```python
+class Team(Enum):
+    DEFENDER = 0
+    CHALLENGER = 1
+
+class Claim:
+    position: uint64
+    claim_hash: ClaimHash
+
+MAX_TRACE = 2**MAX_GAME_DEPTH
+
+def respond(claim: Claim, chal: Team, chal_trace: List[ClaimHash, MAX_TRACE]):
+    if depth(claim.position) % 2 != chal.value:
+        if chal_trace[trace_index(claim.position)] == claim.claim_hash:
+            defend()
+        else:
+            attack()
+    else: pass # no response
+```
+
+In attack or defense, the honest challenger MUST submit a `ClaimHash` corresponding to the state
+identified by the trace index of their response position.
+
+The honest challenger SHOULD respond to claims as soon as possible to avoid the clock of its
+counter-claim from expiring.
+
+### Steps
+
+At the max depth of the game, claims represent commitments to the state of the fault proof VM
+at a single instruction step interval.
+Because the game can no longer bisect further, when the honest challenger has a valid move
+against these claims (valid defined by the response in [Counter Claims](#counter-claims)),
+the only option for an honest challenger is to execute a VM step on-chain to disprove the claim
+at `MAX_GAME_DEPTH`.
+If the VM step proves this claim correct, the claim will be left uncountered.
+
+The same rules for determining whether a move is an attack, defense, or noop from the above
```

Comment:
I think this section has a fair amount of repetition to the one above - it might make sense to combine them so as to articulate from the outset that calling `attack` or `defend` won't work and the challenger should issue the step (with the hyperlink)

Reply:
Will reword. I wanted to highlight that how a challenger's  _stance_ (attack or defense, but not the moves) is determined is the same at `MAX_GAME_DEPTH`. We don't have existing verbiage for this kind of thing. Since we overload `attack` and `defense` in multiple contexts.

Reply:
Ah understood, that makes sense to me!